src/main/java/org/jenkinsci/plugins/workflow/support/steps/WorkspaceStepExecution.java

```diff
@@ -60,12 +65,26 @@ public boolean start() throws Exception {
         flowNode.addAction(new WorkspaceActionImpl(workspace, flowNode));
         listener.getLogger().println("Running in " + workspace);
         body = getContext().newBodyInvoker()
-                .withContext(workspace)
+                .withContexts(
+                    EnvironmentExpander.merge(getContext().get(EnvironmentExpander.class),
+                        new ExpanderImpl(workspace.getRemote())),
+                    workspace)
                 .withCallback(new Callback(lease))
                 .start();
         return false;
     }
 
+    private static final class ExpanderImpl extends EnvironmentExpander {
```

Comment:
Probably I need to introduce a convenience implementation of `EnvironmentExpander` that just expands a predefined map of values (not computed on demand), since a few plugins now do this.